src/S3/CalculatesChecksumTrait.php

```diff
@@ -16,35 +23,36 @@ trait CalculatesChecksumTrait
     public static function getEncodedValue($requestedAlgorithm, $value) {
         $requestedAlgorithm = strtolower($requestedAlgorithm);
         $useCrt = extension_loaded('awscrt');
-        if ($useCrt) {
-            $crt = new Crt();
-            switch ($requestedAlgorithm) {
-                case 'crc32c':
-                    return base64_encode(pack('N*',($crt->crc32c($value))));
-                case 'crc32':
-                    return base64_encode(pack('N*',($crt->crc32($value))));
-                case 'sha256':
-                case 'sha1':
-                    return base64_encode(Psr7\Utils::hash($value, $requestedAlgorithm, true));
-                default:
-                    break;
-                throw new InvalidArgumentException(
-                    "Invalid checksum requested: {$requestedAlgorithm}."
-                    . "  Valid algorithms are CRC32C, CRC32, SHA256, and SHA1."
-                );
+
+        if (isset($requestedAlgorithm, self::$supportedAlgorithms)) {
```

Comment:
Do you want to check if `$requestedAlgorithm` is set in `self::$supportedAlgorithms`, or you want to check if those two variables are set?.

Reply:
If is the first case then maybe you made an unintentional mistake. It should be `isset(self::$supportedAlgorithms[$requestedAlgorithm]);`

Reply:
Whoops.  Rushed through those corrections